### DISCUSSION TITLE
For compile commands what causes the extension to see compile error status?

### DISCUSSION THREAD

**Discussion by m-tyler** [codefori/vscode-ibmi#670] [IBM i]
When I create with CTRSQLRPGI I get a notification about the compile failure.  But if I compile with our custom compile command, which runs our CMS product's compile command, I don't get compile failures.  

What is it that triggers off the notification about the compile being in error?  My custom compile command halts with a job completion error level of 20.  

When my custom compile command ends with level 20 it also causes the CMS product compile to produce a PL/1 source dump (because of the default reply of PL/1 message IDs).  So there is an extra spooled report.  

I can see the extension getting the EVFEVENT data for both methods.

**Comment by worksofliam** [comment]
The extension checks the exit code that is returned from the command. The code can be determined with the `exit` C function.

**Comment by m-tyler** [comment]
Okay, that helps.  Researching further I can see the issue is with my CMS product's compile command that I wrap in my custom compile command.  

I have to use the CMS compile command because it sets the *LIBL correctly which might be a little different than the connection I am currently using.

**Reply by Justin-E-Taylor** [reply]
Have you found anything on this?  We use Alan Campin's COMPILE utility.  When I compile, the IBM i Output view shows the joblog.  Compile errors aren't shown in the Problems view or inline in the source.

**Reply by m-tyler** [reply]
Yes, but at the moment I have to use the standard CRT commands and then I end up using our custom build command when all code compiles clean.

**Reply by worksofliam** [reply]
@Justin-E-Taylor 

> We use Alan Campin's COMPILE utility. When I compile, the IBM i Output view shows the joblog. Compile errors aren't shown in the Problems view or inline in the source.

This is because the command needs to contain `*EVENTF` in the string, which will then fetch the errors into the problems view for you.

**Comment by Justin-E-Taylor** [comment]
Here's the Action I defined (I copied it from a working RDi and changed the variable names):
`COMPILE/COMPILE OBJ(&OPENLIB/&OPENMBR) SRCFILE(&OPENLIB/&OPENSPF) SRCMBR(&OPENMBR) ENV(*GUI) BATCH(*NO)  /* *EVENTF SRCMBR(&N) */`

Here's a simple example I threw together:
```
**free
//    *_> ADDLIBLE LIB(QGPLT) POSITION(*BEFORE QGPL)
//    *_> CRTSQLRPGI OBJ(QGPLT/@4) SRCFILE(@2/@1) SRCMBR(@3) +
//    *_>   OBJTYPE(*MODULE) OPTION(*EVENTF) DBGVIEW(*SOURCE) +
//    *_>   RPGPPOPT(*LVL2) COMPILEOPT('TGTCCSID(*JOB)') CLOSQLCSR(*ENDMOD)
//    *_> CRTPGM PGM(QGPLT/@4)
//    *_> DLTMOD MODULE(QGPLT/@4)
//    *_> RMVLIBLE LIB(QGPLT)
       Ctl-opt
         Debug(*Yes) Option(*SrcStmt: *NoDebugIO :*NoUnref) Optimize(*Basic)
         ExtBinInt(*Yes)
       ;

       x += 1;

       *inLR = *on;
       Return;
```

**Reply by worksofliam** [reply]
@m-tyler Does this actually work in the Action with `/* *EVENTF SRCMBR(&N) */` in the command?

**Reply by m-tyler** [reply]
This is the command I run `WBUILD OBJ(pgmr/pgmnam) SRCFILE(pgmr/srcpf) SRCMBR(pgmsrc) DBGVIEW(*SOURCE) OPTION(*EVENTF) USECMS(*YES)`

This command is a copy of the tool built by Scott Klement, modified to add logic for our CMS product (Rocket LM(i)).  When the LM(i) create command fails, it sets the job status to 0 even though it ends in error.   
 
The issue is with our CMS create tools.  I just needed to figure out what this extension looks for to know why my create action failed.   I have to modify our WBUILD command to compensate for the CMS create commands. 

Until I get time to fix our WBUILD command I have to use the default packed create commands then I can use our command.

**Reply by Justin-E-Taylor** [reply]
> 

The &N is an RDi variable that I forgot to change.  Now that you ask, I don't think that will do anything.  That looks like a CL comment.  I removed it, but the behavior didn't change.  It still runs the compile but fails to bring in the compiler errors.
